# E-Commerce Customer Churn & Retention Analysis
## 04 — Customer Segmentation

### Objective
Combine the strongest churn signals identified in the previous analysis into simple, interpretable customer risk segments.

The segmentation focuses on two of the clearest churn indicators:
- **Customer lifecycle:** customers with tenure of 0–3 months showed substantially higher churn.
- **Complaint history:** customers who registered a complaint showed substantially higher churn.

For this notebook, a **new customer** is defined as a customer with `Tenure <= 3` months. Customers with tenure above 3 months are classified as longer-tenure customers.


## 1. Load Cleaned Dataset

The cleaned dataset is loaded directly so that data preparation steps are not repeated.


In [1]:
import pandas as pd 

In [2]:
df = pd.read_excel("../Data/E Commerce Dataset Cleaned.xlsx")

## 2. Segment Customers by Tenure and Complaint Status

Combining lifecycle stage and complaint status creates four mutually exclusive customer profiles. For each profile, the analysis measures:

- Number of customers
- Number of churned customers
- Churn rate

The filters use combined Boolean conditions rather than chained filtering, avoiding index re-alignment warnings and making each segment definition explicit.


### 2.1 New Customers with a Complaint

Customers with tenure of 0–3 months who have registered a complaint.


In [3]:
df[(df['Tenure'] <= 3) & (df['Complain'] == 1)]['CustomerID'].count()


np.int64(522)

In [4]:
df[(df['Tenure'] <= 3) & (df['Complain'] == 1)]['Churn'].sum()


np.int64(345)

In [5]:
(df[(df['Tenure'] <= 3) & (df['Complain'] == 1)]['Churn'].mean() * 100).round(2)


np.float64(66.09)

**Finding:** This is the highest-risk segment. It contains **522 customers**, of whom **345 churned**, resulting in a **66.09% churn rate**. The combination of being early in the customer lifecycle and having a complaint identifies a particularly vulnerable customer group.


### 2.2 New Customers without a Complaint

Customers with tenure of 0–3 months who have not registered a complaint.


In [6]:
df[(df['Tenure'] <= 3) & (df['Complain'] == 0)]['CustomerID'].count()


np.int64(1038)

In [7]:
df[(df['Tenure'] <= 3) & (df['Complain'] == 0)]['Churn'].sum()


np.int64(308)

In [8]:
(df[(df['Tenure'] <= 3) & (df['Complain'] == 0)]['Churn'].mean() * 100).round(2)


np.float64(29.67)

**Finding:** This segment contains **1,038 customers**, with **308 churned** and a **29.67% churn rate**. Even without a complaint, new customers remain substantially more vulnerable than established customers, reinforcing the importance of the early customer lifecycle.


### 2.3 Longer-Tenure Customers with a Complaint

Customers with tenure above 3 months who have registered a complaint.


In [9]:
df[(df['Tenure'] > 3) & (df['Complain'] == 1)]['CustomerID'].count()


np.int64(1082)

In [10]:
df[(df['Tenure'] > 3) & (df['Complain'] == 1)]['Churn'].sum()


np.int64(163)

In [11]:
(df[(df['Tenure'] > 3) & (df['Complain'] == 1)]['Churn'].mean() * 100).round(2)


np.float64(15.06)

**Finding:** Among **1,082 longer-tenure customers with a complaint**, **163 churned**, producing a **15.06% churn rate**. Complaint history remains relevant, but its association with churn is far less pronounced among established customers than among new customers.


### 2.4 Longer-Tenure Customers without a Complaint

Customers with tenure above 3 months who have not registered a complaint.


In [12]:
df[(df['Tenure'] > 3) & (df['Complain'] == 0)]['CustomerID'].count()


np.int64(2988)

In [13]:
df[(df['Tenure'] > 3) & (df['Complain'] == 0)]['Churn'].sum()


np.int64(132)

In [14]:
(df[(df['Tenure'] > 3) & (df['Complain'] == 0)]['Churn'].mean() * 100).round(2)


np.float64(4.42)

**Finding:** This is the lowest-risk profile. The segment contains **2,988 customers**, with only **132 churned**, resulting in a **4.42% churn rate**.


## 3. Risk Segmentation Summary

The combination of tenure and complaint history produces a clear progression in customer churn risk:

| Customer Segment | Customers | Churned | Churn Rate |
|---|---:|---:|---:|
| New + Complaint | 522 | 345 | **66.09%** |
| New + No Complaint | 1,038 | 308 | **29.67%** |
| Longer Tenure + Complaint | 1,082 | 163 | **15.06%** |
| Longer Tenure + No Complaint | 2,988 | 132 | **4.42%** |

### Retention Priority

**Highest Priority — New customers with complaints**  
This segment combines the two strongest risk characteristics and has by far the highest churn rate.

**Elevated Priority — New customers without complaints**  
Their churn rate remains high despite the absence of a complaint, indicating that early-tenure customers require attention more broadly.

**Lower Priority — Longer-tenure customers with complaints**  
Complaint history is still relevant, but the observed churn rate is substantially lower than for new customers.

**Lowest Observed Risk — Longer-tenure customers without complaints**  
This is the largest segment and has the lowest churn rate.

### Key Insight

The segmentation indicates that **customer tenure and complaint history become especially informative when considered together**. Early-tenure customers represent the main retention-risk population, and a complaint during this early lifecycle is associated with particularly high churn.

These segments provide a simple, interpretable basis for prioritizing retention efforts without introducing unnecessary model complexity.
